src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers.Tests/AZC0009Tests.cs

```diff
@@ -0,0 +1,128 @@
+{
+    public class /*MM*/SomeClientOptions {
+
+        public enum ServiceVersion
+        {
+            V2018_11_09 = 0
+        }
+    }
+}
+");
+            var diagnostics = await _runner.GetDiagnosticsAsync(testSource.Source);
+
+            var diagnostic = Assert.Single(diagnostics);
+
+            Assert.Equal("AZC0009", diagnostic.Id);
+            AnalyzerAssert.DiagnosticLocation(testSource.DefaultMarkerLocation, diagnostics[0].Location);
+        }
+
+        [Fact]
+        public async Task AZC0009ProducedForClientOptionsWithoutServiceVersionInCtor()
+        {
+            var testSource = TestSource.Read(@"
+namespace RandomNamespace
+{
+    public class SomeClientOptions {
+
+        public enum ServiceVersion
+        {
+            V2018_11_09 = 0
+        }
+
+        public /*MM*/SomeClientOptions()
+        {
+        }
+    }
+}
+");
+            var diagnostics = await _runner.GetDiagnosticsAsync(testSource.Source);
+
+            var diagnostic = Assert.Single(diagnostics);
+
+            Assert.Equal("AZC0009", diagnostic.Id);
+            AnalyzerAssert.DiagnosticLocation(testSource.DefaultMarkerLocation, diagnostics[0].Location);
+        }
+
+        [Fact]
+        public async Task AZC0009ProducedForClientOptionsCtorWhereServiceVersionNotFirstParam()
+        {
+            var testSource = TestSource.Read(@"
+namespace RandomNamespace
+{
+    public class SomeClientOptions { 
+
+        public enum ServiceVersion
+        {
+            V2018_11_09 = 0
+        }
+
+        public string AnOption { get; }
+
+        public ServiceVersion Version { get; }
+
+        public SomeClientOptions(string /*MM*/anOption, ServiceVersion version = ServiceVersion.V2018_11_09)
+        {
+            this.AnOption = anOption;
+            this.Version = version;
+        }
+    }
+}
+");
+            var diagnostics = await _runner.GetDiagnosticsAsync(testSource.Source);
+
+            var diagnostic = Assert.Single(diagnostics);
+
+            Assert.Equal("AZC0009", diagnostic.Id);
+            AnalyzerAssert.DiagnosticLocation(testSource.DefaultMarkerLocation, diagnostics[0].Location);
+        }
+
+        [Fact]
+        public async Task AZC0009NotProducedForClientOptionsCtorWhereServiceVersionFirstParam()
+        {
+            var testSource = TestSource.Read(@"
+namespace RandomNamespace
+{
+    public class SomeClientOptions { 
+
+        public enum ServiceVersion
+        {
+            V2018_11_09 = 0
+        }
+
+        public ServiceVersion Version { get; }
+
+        public SomeClientOptions(ServiceVersion version = ServiceVersion.V2018_11_09)
+        {
+            this.Version = version;
+        }
+    }
+}
+");
+            var diagnostics = await _runner.GetDiagnosticsAsync(testSource.Source);
+
+            if (diagnostics == null)
+            {
+                return;
+            }
+
+            Assert.True(diagnostics.Where(d => d.Id == "AZC0009").FirstOrDefault() == null);
```

Comment:
Are other diagnostics produces here?

Reply:
Addressed.  This is again to prevent tests from swallowing unexpected errors?

Reply:
Yes